src/api.cc

```diff
@@ -690,7 +690,10 @@ class LocalLiteInstall : public LiteInstall {
     client_->logTarget("Downloading: ", *target_);
 
     auto downloader = createOfflineDownloader();
+    client_->notifyDownloadStarted(*target_, reason);
```

Comment:
It looks weird to me each time I look into it. What's the point to send a callback to a caller from the method/function invoked by the caller. It may lead to lots of issues in the future...

I think that's fine to have callbacks for the CLI tools and the daemon but not for the API. Unfortunately, it is not easy to implement, but we have to think about it.